src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1573,6 +1576,430 @@ LedgerManagerImpl::prefetchTransactionData(AbstractLedgerTxnParent& ltx,
+                case EXTEND_FOOTPRINT_TTL:
+                    sorobanOpResult.tr().extendFootprintTTLResult().code(
+                        EXTEND_FOOTPRINT_TTL_RESOURCE_LIMIT_EXCEEDED);
+                    break;
+                case RESTORE_FOOTPRINT:
+                    sorobanOpResult.tr().restoreFootprintResult().code(
+                        RESTORE_FOOTPRINT_RESOURCE_LIMIT_EXCEEDED);
+                    break;
+                default:
+                    throw std::runtime_error("unknown soroban op type type");
+                }
+
+                break;
+            }
+        }
+    };
+
+    for (auto const& txBundle : txs)
+    {
+        uint32_t readBytes = 0;
+        getEntries(txBundle.getTx(), txBundle.getResPayload(),
+                   txBundle.getTx()->sorobanResources().footprint.readOnly,
+                   readBytes);
+        getEntries(txBundle.getTx(), txBundle.getResPayload(),
+                   txBundle.getTx()->sorobanResources().footprint.readWrite,
+                   readBytes);
+    }
+
+    return entryMap;
+}
+
+UnorderedMap<LedgerKey, uint32_t>
+LedgerManagerImpl::applyThread(ThreadEntryMap& entryMap, Thread const& thread,
+                               Config const& config,
+                               SorobanNetworkConfig const& sorobanConfig,
+                               ParallelLedgerInfo const& ledgerInfo,
+                               Hash const& sorobanBasePrngSeed,
+                               SorobanMetrics& sorobanMetrics)
+{
+    // RO extensions should only be observed when the entry is modified, so
+    // we accumulate the RO extensions until we need to apply them.
+    UnorderedMap<LedgerKey, uint32_t> readOnlyTtlExtensions;
+
+    for (auto const& txBundle : thread)
+    {
+        releaseAssertOrThrow(txBundle.getResPayload());
+
+        SHA256 txSubSeedSha;
+        txSubSeedSha.add(sorobanBasePrngSeed);
+        txSubSeedSha.add(xdr::xdr_to_opaque(txBundle.getTxNum()));
+        Hash txSubSeed = txSubSeedSha.finish();
+
+        auto const& readWrite =
+            txBundle.getTx()->sorobanResources().footprint.readWrite;
+        for (auto const& lk : readWrite)
+        {
+            if (!isSorobanEntry(lk))
+            {
+                continue;
+            }
+
+            auto const& ttlKey = getTTLKey(lk);
+            auto extIt = readOnlyTtlExtensions.find(ttlKey);
+            if (extIt != readOnlyTtlExtensions.end() && extIt->second != 0)
+            {
+                // "Commit" all RO cumulative bumps now that the key is in a
+                // readWrite set
+                auto it = entryMap.find(ttlKey);
+                if (it != entryMap.end())
+                {
+                    releaseAssertOrThrow(it->second.mLedgerEntry);
+
+                    it->second.mLedgerEntry->data.ttl().liveUntilLedgerSeq =
+                        std::min(
+                            sorobanConfig.stateArchivalSettings().maxEntryTTL,
```

Comment:
I believe you should do `current + min(maxTTL, cumulativeTTL)` instead

Reply:
This has been removed